pipe-cli/src/utilities/dts_operations_manager.py

```diff
@@ -0,0 +1,99 @@
+# Copyright 2017-2021 EPAM Systems, Inc. (https://www.epam.com/)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import click
+import json
+import sys
+from prettytable import prettytable
+
+from src.api.dts import DTS
+from src.model.dts_model import DtsEncoder
+
+
+class DtsOperationsManager:
+
+    _DTS_TABLE_HEADERS = ['Registry ID', 'Name', 'URL', 'Created date', 'Schedulable', 'Prefixes', 'Preferences']
+    _PREF_DELIMITER = "="
+
+    def __init__(self):
+        pass
+
+    def create(self, url, name, schedulable, prefixes, preferences, json_out):
+        registry = DTS.create(url, name, schedulable, prefixes, self._convert_preferences_list_to_dict(preferences))
+        self._print_registry(registry, json_out)
+
+    def list(self, registry_id, json_out):
+        if registry_id:
+            registry = DTS.load(registry_id)
+            self._print_registry(registry, json_out)
+        else:
+            registries = DTS.load_all()
+            if json_out:
+                self._print_dts_json(registries)
+            else:
+                self._print_registries_prettytable(registries)
+
+    def upsert_preferences(self, registry_id, preferences_list, json_out):
+        if not preferences_list:
+            click.echo('Preferences should not be empty!', err=True)
+            sys.exit(1)
+        updated_registry = DTS.update_preferences(registry_id, self._convert_preferences_list_to_dict(preferences_list))
+        self._print_registry(updated_registry, json_out)
+
+    def delete_preferences(self, registry_id, preferences_keys, json_out):
+        if not preferences_keys:
+            click.echo('Preferences keys to be removed should not be empty!', err=True)
+            sys.exit(1)
+        updated_registry = DTS.delete_preferences(registry_id, preferences_keys)
+        self._print_registry(updated_registry, json_out)
+
+    def _convert_preferences_list_to_dict(self, preferences_list):
+        preferences_dict = {}
+        for preference_entry in preferences_list:
+            preference_value_and_key = preference_entry.split(self._PREF_DELIMITER)
+            if len(preference_value_and_key) != 2:
+                click.echo('Error [%s]: preference should contain exactly one delimiter!' % preference_entry, err=True)
+                sys.exit(1)
+            else:
+                preferences_dict[preference_value_and_key[0]] = preference_value_and_key[1]
```

Comment:
Let's split by the first delimiter occurrence and as a result support values with `self._PREF_DELIMITER` in it.

```python
preference_value_and_key = preference_entry.split(self._PREF_DELIMITER, 1)
```